hw/ip/otbn/dv/otbnsim/sim/state.py

```diff
@@ -540,10 +541,30 @@ def stop_at_end_of_cycle(self, err_bits: int) -> None:
         Any bits set in err_bits will be set in the ERR_BITS register when
         we're done.
 
+        Some errors are delayed by one cycle to match the RTL's behaviour.
         '''
+        # Delay certain errors due to the registering of escalation signals
+        # - The DMEM_INTG_VIOLATION error is raised by a LW or BN.LID instruction if the returned
+        #   data is invalid. If this is the case in RTL, the wrong data is still commited into the
+        #   register file and the escalation takes place in the next cycle.
+        if err_bits & ErrBits.DMEM_INTG_VIOLATION:
+            err_bits &= ~ErrBits.DMEM_INTG_VIOLATION
+            self._pending_err_bits |= ErrBits.DMEM_INTG_VIOLATION
+            # Do not stop if there are no other error bits set
```

Comment:
To avoid the `return` here, is it true that we want to cause a new stop iff `self._err_bits` ends up nonzero? If so, maybe it would be cleaner to change the last line to be `self.pending_halt |= (self._err_bits != 0)` instead?